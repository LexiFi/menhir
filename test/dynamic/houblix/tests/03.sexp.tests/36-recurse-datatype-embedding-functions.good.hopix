type s = C (int, unit -> s)

fun : int -> s
      from (n : int) = C<> (n, (\(u : unit) -> from (n + 1) : unit -> s))