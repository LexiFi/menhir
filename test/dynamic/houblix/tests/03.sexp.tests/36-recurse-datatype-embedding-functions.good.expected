(((value
    (DefineType
      ((value (TCon s)) (position ((start_p ("" 1 0 5)) (end_p ("" 1 0 6)))))
      ()
      (DefineSumType
        ((((value (KId C))
            (position ((start_p ("" 1 0 9)) (end_p ("" 1 0 10)))))
           (((value (TyCon (TCon int) ()))
              (position ((start_p ("" 1 0 12)) (end_p ("" 1 0 15)))))
             ((value
                (TyArrow
                  ((value (TyCon (TCon unit) ()))
                    (position ((start_p ("" 1 0 17)) (end_p ("" 1 0 21)))))
                  ((value (TyCon (TCon s) ()))
                    (position ((start_p ("" 1 0 25)) (end_p ("" 1 0 26)))))))
               (position ((start_p ("" 1 0 17)) (end_p ("" 1 0 26)))))))))))
   (position ((start_p ("" 1 0 0)) (end_p ("" 1 0 27)))))
  ((value
     (DefineValue
       (RecFunctions
         ((((value (Id from))
             (position ((start_p ("" 4 44 50)) (end_p ("" 4 44 54)))))
            (((value
                (ForallTy ()
                  ((value
                     (TyArrow
                       ((value (TyCon (TCon int) ()))
                         (position
                           ((start_p ("" 3 29 35)) (end_p ("" 3 29 38)))))
                       ((value (TyCon (TCon s) ()))
                         (position
                           ((start_p ("" 3 29 42)) (end_p ("" 3 29 43)))))))
                    (position ((start_p ("" 3 29 35)) (end_p ("" 3 29 43)))))))
               (position ((start_p ("" 3 29 34)) (end_p ("" 3 29 43))))))
            (FunctionDefinition
              ((value
                 (PTypeAnnotation
                   ((value
                      (PVariable
                        ((value (Id n))
                          (position
                            ((start_p ("" 4 44 56)) (end_p ("" 4 44 57)))))))
                     (position ((start_p ("" 4 44 56)) (end_p ("" 4 44 57)))))
                   ((value (TyCon (TCon int) ()))
                     (position ((start_p ("" 4 44 60)) (end_p ("" 4 44 63)))))))
                (position ((start_p ("" 4 44 55)) (end_p ("" 4 44 64)))))
              ((value
                 (Tagged
                   ((value (KId C))
                     (position ((start_p ("" 4 44 67)) (end_p ("" 4 44 68)))))
                   (())
                   (((value
                       (Variable
                         ((value (Id n))
                           (position
                             ((start_p ("" 4 44 72)) (end_p ("" 4 44 73)))))
                         ()))
                      (position
                        ((start_p ("" 4 44 72)) (end_p ("" 4 44 73)))))
                     ((value
                        (TypeAnnotation
                          ((value
                             (Fun
                               (FunctionDefinition
                                 ((value
                                    (PTypeAnnotation
                                      ((value
                                         (PVariable
                                           ((value (Id u))
                                             (position
                                               ((start_p ("" 4 44 78))
                                                 (end_p ("" 4 44 79)))))))
                                        (position
                                          ((start_p ("" 4 44 78))
                                            (end_p ("" 4 44 79)))))
                                      ((value (TyCon (TCon unit) ()))
                                        (position
                                          ((start_p ("" 4 44 82))
                                            (end_p ("" 4 44 86)))))))
                                   (position
                                     ((start_p ("" 4 44 77))
                                       (end_p ("" 4 44 87)))))
                                 ((value
                                    (Apply
                                      ((value
                                         (Variable
                                           ((value (Id from))
                                             (position
                                               ((start_p ("" 4 44 91))
                                                 (end_p ("" 4 44 95)))))
                                           ()))
                                        (position
                                          ((start_p ("" 4 44 91))
                                            (end_p ("" 4 44 95)))))
                                      ((value
                                         (Apply
                                           ((value
                                              (Apply
                                                ((value
                                                   (Variable
                                                     ((value (Id `+`))
                                                       (position
                                                         ((start_p
                                                            ("" 4 44 99))
                                                           (end_p
                                                             ("" 4 44 100)))))
                                                     ()))
                                                  (position
                                                    ((start_p ("" 4 44 99))
                                                      (end_p ("" 4 44 100)))))
                                                ((value
                                                   (Variable
                                                     ((value (Id n))
                                                       (position
                                                         ((start_p
                                                            ("" 4 44 97))
                                                           (end_p
                                                             ("" 4 44 98)))))
                                                     ()))
                                                  (position
                                                    ((start_p ("" 4 44 97))
                                                      (end_p ("" 4 44 98)))))))
                                             (position
                                               ((start_p ("" 4 44 97))
                                                 (end_p ("" 4 44 100)))))
                                           ((value
                                              (Literal
                                                ((value (LInt 1))
                                                  (position
                                                    ((start_p ("" 4 44 101))
                                                      (end_p ("" 4 44 102)))))))
                                             (position
                                               ((start_p ("" 4 44 101))
                                                 (end_p ("" 4 44 102)))))))
                                        (position
                                          ((start_p ("" 4 44 96))
                                            (end_p ("" 4 44 103)))))))
                                   (position
                                     ((start_p ("" 4 44 91))
                                       (end_p ("" 4 44 103))))))))
                            (position
                              ((start_p ("" 4 44 76)) (end_p ("" 4 44 103)))))
                          ((value
                             (TyArrow
                               ((value (TyCon (TCon unit) ()))
                                 (position
                                   ((start_p ("" 4 44 106))
                                     (end_p ("" 4 44 110)))))
                               ((value (TyCon (TCon s) ()))
                                 (position
                                   ((start_p ("" 4 44 114))
                                     (end_p ("" 4 44 115)))))))
                            (position
                              ((start_p ("" 4 44 106)) (end_p ("" 4 44 115)))))))
                       (position
                         ((start_p ("" 4 44 75)) (end_p ("" 4 44 116))))))))
                (position ((start_p ("" 4 44 67)) (end_p ("" 4 44 117)))))))))))
    (position ((start_p ("" 3 29 29)) (end_p ("" 4 44 117))))))
